packages/core/src/interpreter.ts

```diff
@@ -651,26 +651,16 @@ export class Interpreter<
             return;
           }
           ref._parent = this; // TODO: fix
-          // If the actor will be stopped right after it's started
-          // (such as in transient states) don't bother starting the actor.
-          if (
-            state.actions.find((otherAction) => {
-              return (
-                otherAction.type === actionTypes.stop &&
-                (otherAction as StopActionObject).params.actor === id
-              );
-            })
-          ) {
+          // If the actor didn't end up being in the state
+          // (eg. going through transient states could stop it) don't bother starting the actor.
+          if (!this.state.children[id]) {
```

Comment:
Since `state.children` are mutated in `setChildren` "in order" it should be enough to just check if the final result contains the given id or not.